### PR TITLE
filter: add label for filterCounter

### PR DESCRIPTION
**Pull request by Yisaer**
### What is changed and how it works?

Add label for `filterCounter`. If the `Filter` filter the `target` by comparing origin store and target store, we can record the origin store id into metrics.

Then we can query the metrics like this:

image

By this way, we can easily judge whether the filter is reasonable.
 

### Check List

Tests

- Unit test

### Release note

* No release note

### CODE REVIEW

**Comment by Yisaer** [comment]
@rleungx  PTAL

**Comment by lhy1024** [comment]
LGTM

File: server/schedule/filter/filters.go [hunk 1]
```diff
@@ -76,13 +87,23 @@ type Filter interface {
 	Target(opt *config.PersistOptions, store *core.StoreInfo) bool
 }
 
+// ComparingFilter is an interface to filter target store by comparing source and target stores
+type ComparingFilter interface {
+	Filter
+	// GetOldStoreID returns the source store when comparing.
+	GetOldStoreID() uint64
```
Comment (rleungx) [inline server/schedule/filter/filters.go:94]: Why not source store

Comment (Yisaer) [reply server/schedule/filter/filters.go]: updated.

File: server/schedule/filter/filters.go [hunk 2]
```diff
@@ -166,6 +194,7 @@ type distinctScoreFilter struct {
 	stores    []*core.StoreInfo
 	policy    string
 	safeScore float64
+	oldStore  uint64
```
Comment (rleungx) [inline server/schedule/filter/filters.go:197]:
```suggestion
	srcStore  uint64
```

Comment (Yisaer) [reply server/schedule/filter/filters.go]: updated.

**Comment by rleungx** [comment]
BTW, I think we can add the metrics but hide it

**Comment by lhy1024** [comment]
> BTW, I think we can add the metrics but hide it

In some cases, we use tools to export metrics, which may not be convenient for later troubleshooting after the hidden monitoring

**Comment by Yisaer** [comment]
> > BTW, I think we can add the metrics but hide it
> 
> In some cases, we use tools to export metrics, which may not be convenient for later troubleshooting after the hidden monitoring

We can manually display the hidden panel, then expose the metrics.

**Review by rleungx — APPROVED** [review]

File: metrics/grafana/pd.json [hunk 3]
```diff
@@ -6226,6 +6226,30 @@
               "metric": "pd_scheduler_event_count",
               "refId": "A",
               "step": 4
+            },
+            {
+              "expr": "sum(delta(pd_schedule_filter{action=\"filter-target\",type=\"distinct-filter\"}[1m])) by (source, target, type, scope)",
+              "format": "time_series",
+              "hide": true,
+              "intervalFactor": 2,
+              "legendFormat": "{{scope}}-store-{{source}}-{{type}}-target-{{target}}",
```
Comment (rleungx) [inline metrics/grafana/pd.json:6235]:
```suggestion
              "legendFormat": "{{scope}}-{{type}}-{{source}}-{{target}}",
```

File: server/schedule/filter/filters.go [hunk 4]
```diff
@@ -455,28 +490,33 @@ func (f *ruleFitFilter) Source(opt *config.PersistOptions, store *core.StoreInfo
...
+	scope          string
+	fitter         RegionFitter
+	region         *core.RegionInfo
+	oldFit         *placement.RegionFit
+	srcLeaderStore uint64
 }
 
 // newRuleLeaderFitFilter creates a filter that ensures after transfer leader with new store,
 // the isolation level will not decrease.
 func newRuleLeaderFitFilter(scope string, fitter RegionFitter, region *core.RegionInfo, oldLeaderStoreID uint64) Filter {
```
Comment (rleungx) [inline server/schedule/filter/filters.go:513]:
```suggestion
func newRuleLeaderFitFilter(scope string, fitter RegionFitter, region *core.RegionInfo, srcLeaderStoreID uint64) Filter {
```

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**